src/ingest_validation_tools/plugin_validator.py

```diff
@@ -71,31 +72,35 @@ def run_plugin_validators_iter(metadata_path: PathOrStr,
             if any(row['assay_type'] != assay_type for row in rows):
                 raise ValidatorError(f'{metadata_path} contains more than one assay type')
 
-            if all('data_path' in row for row in rows):
-                for row in rows:
-                    data_path = Path(row['data_path'])
-                    if not data_path.is_absolute():
-                        data_path = (metadata_path / data_path).resolve()
-                    for k, v in validation_error_iter(data_path, assay_type, plugin_dir):
-                        yield k, v
-            else:
-                raise ValidatorError(f'{metadata_path} has no "data_path" column')
         else:
             raise ValidatorError(f'{metadata_path} has no "assay_type" column')
     else:
         raise ValidatorError(f'{metadata_path} does not exist or is not a file')
 
+    if all('data_path' in row for row in rows):
+        for row in rows:
+            data_path = Path(row['data_path'])
+            if not data_path.is_absolute():
+                data_path = (metadata_path / data_path).resolve()
+            for k, v in validation_error_iter(data_path, assay_type, plugin_dir):
+                yield k, v
+    else:
+        raise ValidatorError(f'{metadata_path} has no "data_path" column')
```

Comment:
Preference to rely on these checks being done at a higher level, and only run these checks when the higher level checks pass, and not to add exceptions we don't recover from here. If we did hit this, it would be a bug, and I'm not sure that these `raises` make it easier to debug.

Let's not belabor this now, but where I'm coming from might make more sense when you get to the `bad` test fixtures.

Reply:
Yeah, we have a legitimate disagreement on this point.  My counterargument is that the additional test at this level makes the code easier to read and locally safe, which is always better than relying on a global condition which is not locally verified for code safety.  And I don't see how this 'raise' makes things *harder* to debug!  I agree that we should potentially revisit this when I've dealt with the test fixtures.